pkg/api/vlabs/validate.go

```diff
@@ -509,11 +509,6 @@ func (a *Properties) validateZones() error {
 }
 
 func (a *Properties) validateLinuxProfile() error {
-	for _, publicKey := range a.LinuxProfile.SSH.PublicKeys {
```

Comment:
do we still need to remove this since you moved validation to after auto-fill?

Reply:
I removed it because it's a condition that can never be fulfilled. If the key value was an empty string, a key was generated. Any failure in generating/saving the key throws a fatal error. So you can't get here with an empty key string.

I can put it back if you like, it's just not testing anything.

Reply:
I think we still want it. We also run validate on upgrade/scale and this is also to guard against users making manual alterations to their apimodel before running upgrade/scale (like removing the key).